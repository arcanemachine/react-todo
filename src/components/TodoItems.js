import React from 'react';
import TodoItem from './TodoItem.js';

export default class TodoItems extends React.Component {
  render() {
    const todos = this.props.todos.length ? (
      this.props.todos.map(
        (todo) => <TodoItem key={todo.id}
                            todo={todo}
                            className="mt-4"
                            emitTodoUpdate={this.props.emitTodoUpdate}
                            emitTodoDelete={this.props.emitTodoDelete} />
      )
    ) : (
      <li className="my-4">
        Your todo list is empty.
      </li>
    );
    return (
      <div className="has-text-centered">
        <ul className="my-4 py-2 box">
          {todos}
        </ul>
        <div className="my-4">
          You can modify/remove an item by clicking on it.
        </div>
      </div>
    );
  }
}
