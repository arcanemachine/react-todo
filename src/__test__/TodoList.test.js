import React from 'react';
import ReactDOM from 'react-dom';
import TodoItems from '../components/TodoItems';
import { create } from 'react-test-renderer';
import { render, fireEvent, getByTestId } from 'react-testing-library';
import App from '../App';

describe.skip('My first snapshot test', () => {
  test('Page renders correctly', () => {
    let tree = create(<App />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})
