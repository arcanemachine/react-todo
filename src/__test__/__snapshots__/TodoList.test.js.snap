// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first snapshot test Page renders correctly 1`] = `
Array [
  <form
    className="mt-4 mx-auto is-flex is-justify-content-center"
    onSubmit={[Function]}
  >
    <input
      className="input is-inline is-medium btn-group-left-side"
      onChange={[Function]}
      placeholder="Enter new todo item..."
      type="text"
      value=""
    />
    <input
      className="button is-success is-medium btn-group-right-side"
      type="submit"
      value="Create"
    />
  </form>,
  <div
    className="has-text-centered"
  >
    <ul
      className="my-4 py-2 box"
    >
      <li
        className="my-4"
      >
        Your todo list is empty.
      </li>
    </ul>
    <div
      className="my-4"
    >
      You can modify or remove an item by clicking on it.
    </div>
  </div>,
]
`;
